PetShare/Services/Implementations/Pagination/PaginationService.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Microsoft.Extensions.Options;
+using PetShare.Configuration;
+using PetShare.Models;
+using PetShare.Services.Interfaces.Pagination;
+
+namespace PetShare.Services.Implementations.Pagination
+{
+    public class PaginationService : IPaginationService
+    {
+        private readonly IOptions<PaginationConfiguration> _config;
+        public PaginationService(IOptions<PaginationConfiguration> config)
+        {
+            _config = config;
+        }
+
+        public PaginatedResult<T>? GetPage<T>(IEnumerable<T> collection, PaginationQueryRequest query)
+        {
+            int pageSize = query.PageCount ?? _config.Value.DefaultPageSize;
+            int pageNr = query.PageNumber ?? _config.Value.DefaultPageNumber;
+
+            if (pageNr != 0 && collection.Count() <= pageNr * pageSize)
+                return null;
```

Comment:
Hmmm, IDK if we should treat this as an error. We could return an empty page as well.

Reply:
We should also check if `PageNumber` and `PageCount` are `>0`. IMO using `Result` there would be nice, since we can return `InvalidOperation` with a description of what went wrong.